students/Kollii/lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,78 @@
+from operator import itemgetter
+
+
+class Donor(object):
+   
+    def __init__(self, name, donation_amt=None):
+        """ Initializes the donor object """
+        self.name = name
+        self.donation_amt = donation_amt if donation_amt is not None else []
+
+    def add_donation(self, amount):
+        self.donation_amt.append(amount)
+
+
+    def print_thanksnote(self):
+        note = (f'\nDear {self.name},'
+                    f'\nThank you for your generous donation total amount of ${self.donation_amt[-1]:.2f}.'
+                    '\nWe appreciate your support.'
+                    '\n\nRegards,\nXYZ Charity Team\n')
+        return note
+
+    def sort_key(self):
+        return sum(self.donation_amt)
+
+class DonorCollection():
+    """ Initialize donors dictionary """
+    def __init__(self):
+        self.donors = {}
+
+    @staticmethod
+    def initialize_dict(donor_data):
+        # create a new collection object
+        donors = DonorCollection()                     
+        for name, amounts in donor_data.items():     
+            donor = Donor(name, amounts)            
+            # add the new donor to the collection
+            donors.add(donor)
+        return donors  
+
+    def add(self, donor):
+        """ Add donor to the collection """
+        self.donors[donor.name] = donor
+
+    def donors_list(self):
+        """ Generate a list of donors """
+        ls=[]
+        for names in self.donors:
+            ls.append(names)
+        return '\n'.join(ls)
```

Comment:
you can use a list comprehension here - however, it's even simpler:
`return '\n'.join(list(self.donors))`